mmv1/templates/terraform/constants/firewall.tmpl

```diff
@@ -107,6 +107,30 @@ func diffSuppressSourceRanges(k, old, new string, d *schema.ResourceData) bool {
 			return true
 		}
 	}
-	// For any other source_ranges value diff, don't suppress
-	return false
+
+	return diffSuppressCIDR("source_ranges", d)
+}
+
+func diffSuppressDestinationRanges(k, old, new string, d *schema.ResourceData) bool {
+	return diffSuppressCIDR("destination_ranges", d)
+}
+
+func diffSuppressCIDR (key string, d *schema.ResourceData) bool {
+	o, n := d.GetChange(key)
+	oldRanges := o.(*schema.Set).List()
+	newRanges := n.(*schema.Set).List()
+
+	if len(oldRanges) != len(newRanges) {
+		return false
+	}
+
+	for i := 0; i < len(oldRanges); i++ {
+		oldCIDR := netip.MustParsePrefix(oldRanges[i].(string))
+		newCIDR := netip.MustParsePrefix(newRanges[i].(string))
+
+		if oldCIDR != newCIDR {
+			return false
+		}
+	}
```

Comment:
[Comment]

Not a contributor, just happened to hit this issue and excited to find someone already working on a fix 😅 

I noticed they use common helpers for this elsewhere: https://github.com/GoogleCloudPlatform/magic-modules/blob/fd62c38302087df4545b6a3496df8aef38a9e02a/tpgtools/ignored_handwritten/common_diff_suppress.go#L58-L64

Might want to consider adopting/expanding that.

Thank you for working on this!

Reply:
We generally try not to add diff suppress functions to the common library unless there's strong evidence that the behavior will be the same across at least a few different API surfaces. I'm not sure offhand whether that's the case here, so I'd be inclined to leave it as specific to `compute` for now.